l1-contracts/test/foundry/l2/integration/L2Utils.sol

```diff
@@ -145,25 +191,48 @@ library L2Utils {
                 address(0),
                 ethAssetId
             )
+        );
+    }
+
+    function forceDeployWithConstructor(
+        string memory _contractName,
+        address _address,
+        bytes memory _constructorArgs
+    ) public {
+        bytes memory bytecode = readEraBytecode(_contractName);
+
+        bytes32 bytecodehash = L2ContractHelper.hashL2Bytecode(bytecode);
+
+        IContractDeployer.ForceDeployment[] memory deployments = new IContractDeployer.ForceDeployment[](1);
+        deployments[0] = IContractDeployer.ForceDeployment({
+            bytecodeHash: bytecodehash,
+            newAddress: _address,
+            callConstructor: true,
+            value: 0,
+            input: _constructorArgs
         });
 
         vm.prank(L2_FORCE_DEPLOYER_ADDR);
         IContractDeployer(DEPLOYER_SYSTEM_CONTRACT).forceDeployOnAddresses(deployments);
     }
 
-    /// @notice Encodes the token data.
-    /// @param name The name of the token.
-    /// @param symbol The symbol of the token.
-    /// @param decimals The decimals of the token.
-    function encodeTokenData(
-        string memory name,
-        string memory symbol,
-        uint8 decimals
-    ) internal pure returns (bytes memory) {
-        bytes memory encodedName = abi.encode(name);
-        bytes memory encodedSymbol = abi.encode(symbol);
-        bytes memory encodedDecimals = abi.encode(decimals);
-
-        return abi.encode(encodedName, encodedSymbol, encodedDecimals);
+    function deployViaCreat2L2(
```

Comment:
are you sure this function works?

maybe it started working as a feature of foundry-zksync, but generally for zkEVM `bytecode` is not concatenation of creationCode and constructor args

Reply:
The tests seem to run.. In particular, the L2GatewayTests calls the CTM, Admin facets:
<img width="1068" alt="image" src="https://github.com/user-attachments/assets/444f8bb9-0ada-49be-a8ba-b30407eda707">
<img width="1158" alt="image" src="https://github.com/user-attachments/assets/e23e37e6-9daf-43a7-b8ae-166007bff72f">